pkg/postage/listener/listener_test.go

```diff
@@ -0,0 +1,446 @@
+			),
+		)
+		listener := listener.New(logging.New(ioutil.Discard, 0), mf, postageStampAddress, priceOracleAddress)
+		err := listener.Listen(0, ev)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		select {
+		case e := <-evC:
+			e.(createArgs).compare(t, c) // event args should be equal
+		case <-time.After(5 * time.Second):
+			t.Fatal("timed out waiting for event")
+		}
+
+		select {
+		case e := <-evC:
+			e.(topupArgs).compare(t, topup) // event args should be equal
+		case <-time.After(5 * time.Second):
+			t.Fatal("timed out waiting for event")
+		}
+
+		select {
+		case e := <-evC:
+			e.(depthArgs).compare(t, depthIncrease) // event args should be equal
+		case <-time.After(5 * time.Second):
+			t.Fatal("timed out waiting for event")
+		}
+
+		select {
+		case e := <-evC:
+			e.(priceArgs).compare(t, priceUpdate)
+		case <-time.After(5 * time.Second):
+			t.Fatal("timed out waiting for event")
+		}
+	})
+}
+
+func newEventUpdaterMock() (*updater, chan interface{}) {
+	c := make(chan interface{})
+	return &updater{
+		eventC: c,
+	}, c
+}
+
+type updater struct {
+	eventC chan interface{}
+}
+
+func (u *updater) Create(id, owner []byte, amount, normalisedAmount *big.Int, depth uint8) error {
+	u.eventC <- createArgs{
+		id:               id,
+		owner:            owner,
+		amount:           amount,
+		normalisedAmount: normalisedAmount,
+		depth:            depth,
+	}
+	return nil
+}
+
+func (u *updater) TopUp(id []byte, amount, normalisedBalance *big.Int) error {
+	u.eventC <- topupArgs{
+		id:                id,
+		amount:            amount,
+		normalisedBalance: normalisedBalance,
+	}
+	return nil
+}
+
+func (u *updater) UpdateDepth(id []byte, depth uint8, normalisedBalance *big.Int) error {
+	u.eventC <- depthArgs{
+		id:                id,
+		depth:             depth,
+		normalisedBalance: normalisedBalance,
+	}
+	return nil
+}
+
+func (u *updater) UpdatePrice(price *big.Int) error {
+	u.eventC <- priceArgs{price}
+	return nil
+}
+
+type mockFilterer struct {
+	filterLogEvents    []types.Log
+	subscriptionEvents []types.Log
+	sub                *sub
+}
+
+func newMockFilterer(opts ...Option) *mockFilterer {
+	mock := new(mockFilterer)
+	for _, o := range opts {
+		o.apply(mock)
+	}
+	return mock
+}
+
+func WithFilterLogEvents(events ...types.Log) Option {
+	return optionFunc(func(s *mockFilterer) {
+		s.filterLogEvents = events
+	})
+}
+
+func (m *mockFilterer) FilterLogs(ctx context.Context, query ethereum.FilterQuery) ([]types.Log, error) {
+	return m.filterLogEvents, nil
+}
+
+func (m *mockFilterer) SubscribeFilterLogs(ctx context.Context, query ethereum.FilterQuery, ch chan<- types.Log) (ethereum.Subscription, error) {
+	go func() {
+		for _, ev := range m.subscriptionEvents {
+			ch <- ev
+		}
+	}()
+	s := newSub()
+	return s, nil
+}
+
+func (m *mockFilterer) Close() {
+	close(m.sub.c)
+}
+
+func (m *mockFilterer) BlockNumber(context.Context) (uint64, error) {
+	return 0, nil
+}
+
+func parseABI(json string) abi.ABI {
```

Comment:
you could probably just reuse the same function from `listener.go` and export it using `export_test`

Reply:
i removed this and am just reusing the already parsed abi through export